tests/components/electric_kiwi/test_sensor.py

```diff
@@ -0,0 +1,94 @@
+"""The tests for Electric Kiwi sensors."""
+
+from datetime import timezone
+from unittest.mock import AsyncMock, Mock, patch
+
+from freezegun import freeze_time
+import pytest
+
+from homeassistant.components.electric_kiwi import (
+    DOMAIN,
+    ElectricKiwiHOPDataCoordinator,
+)
+from homeassistant.components.electric_kiwi.const import ATTRIBUTION
+from homeassistant.components.electric_kiwi.sensor import (
+    _check_and_move_time,
+)
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import ATTR_ATTRIBUTION, ATTR_DEVICE_CLASS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_registry import EntityRegistry
+import homeassistant.util.dt as dt_util
+
+from .conftest import TIMEZONE
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize(
+    ("sensor", "sensor_state"),
+    [
+        ("sensor.hour_of_free_power_start", "4:00 PM"),
+        ("sensor.hour_of_free_power_end", "5:00 PM"),
+    ],
+)
+async def test_hop_sensors(
```

Comment:
These tests are not that easy to understand what they does or how it operates.
Can't we just use the `freezer` fixture and go to a certain time and assert the proper state and possibly attributes.
[Example](https://github.com/home-assistant/core/blob/dev/tests/components/workday/test_binary_sensor.py#L69)

Reply:
So what's going on is there is a daily fixed time, think of it like an alarm, it goes off at 4 pm daily and ends at 5pm daily. 

This time (note no day is given, its only a time) is fed from the Electric Kiwi API.

So freezing time wouldn't change the test or the entity state other than the rolling forward one day which is tested separately in the other function test for _check_and_move_time.

in short the sensors should reflect the API 1:1 if the API returns 4:00 PM, the sensor state should be set to today at 4pm or if now is past 4pm, then tomorrow at 4pm

Should I add some more comments to explain what is going on?

Reply:
one alternative is to freeze the time in the past and future of the date to test the correctness of it, this would also in turn test the _check_and_move_time function fully.

Reply:
If you prefer we can leave this test with a bit more comments in it to explain what's happening.

Then simply add another test with freezer with which you can also move time and with `async_fire_time_changed()` you can trigger refresh by the coordinator. By doing this we mimic reality as much as possible which is not really the case with the current test

Reply:
Ok I'm happy with that.

Reply:
Added extra explanation. Let me know if its enough/understandable?